crates/provider/src/shared/types.rs

```diff
@@ -0,0 +1,172 @@
+use std::{
+    collections::BTreeMap, os::unix::process::ExitStatusExt, path::PathBuf, process::ExitStatus,
+    vec,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq)]
+enum ZombieRole {
+    Temp,
+    Node,
+    BootNode,
+    Collator,
+    CumulusCollator,
+    Authority,
+    FullNode,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum PortName {
+    Prometheus,
+    Rpc,
+    RpcWs,
+    P2P,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ImagePullPolicy {
+    IfNotPresent,
+    Never,
+    Always,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FileMap {
+    local_file_path:  PathBuf,
+    remote_file_path: PathBuf,
+    unique:           bool,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RunCommandResponse {
+    pub exit_code: ExitStatus,
+    pub std_out:   Vec<u8>,
+    pub std_err:   Option<Vec<u8>>,
```

Comment:
Why don't use `String`/Option<String>` here and convert from the output of the process?

Reply:
The type of std::process::Output is : `pub stderr: Vec<u8, Global>`;

Isn't it easier to use that type instead of getting into conversions?

Reply:
I think it's better to have a String, that way, when we use a created ```RunCommandResponse.std_err``` we don't need to use a conversion first to use it because it was done at instanciation